tests/repo_finder/test_commit_finder.py

```diff
@@ -34,14 +25,86 @@ def test_get_commit_from_version() -> None:
         "78.109",  # To match a tag separated by characters 'r' 'rv'.
         "1.0.5-JRE",  # To NOT match the similar tag without the 'JRE' suffix.
     ]
+
+    tags = ["test-name-v1.0.1-A", "v1.0.3+test", "v_1.0.5", "50_0_2", "r78rv109", "1.0.5-JRE"]
+
+    # Perform tests
     purl_name = "test-name"
     for count, value in enumerate(versions):
-        _test_version(git_obj, purl_name, value, hash_targets[count])
+        _test_version(tags, purl_name, value, tags[count])
         purl_name = "test-name" + "-" + str(count + 1)
 
 
-def _test_version(git_obj: Git, name: str, version: str, hash_target: str) -> None:
-    """Retrieve commit matching version and check commit hash is correct."""
-    branch, digest = commit_finder.get_commit_from_version(git_obj, name, version)
-    assert branch
-    assert git_obj.get_commit(digest).hash == hash_target
+def _test_version(tags: list[str], name: str, version: str, target_tag: str) -> None:
+    """Retrieve tag matching version and check it is correct."""
+    matched_tags = commit_finder.match_tags(tags, name, version)
+    assert matched_tags
+    assert matched_tags[0] == target_tag
+
+
+input_pattern = re.compile(r"[0-9]{1,3}(\.[0-9a-z]{1,3}){,5}([-+#][a-z0-9].+)?", flags=re.IGNORECASE)
+
+
+@given(hypothesis.strategies.text())
```

Comment:
Is there any reason why we have `hypothesis.strategies.text()` here while other places in this file use the strategy in a different manner? For example on line 71
```python
from hypothesis.strategies import DataObject, data

@given(data())
@settings(max_examples=VERSION_ITERATIONS, deadline=None)
```